folio-spring-i18n/src/main/java/org/folio/spring/i18n/service/TranslationService.java

```diff
@@ -0,0 +1,352 @@
+   *
+   * <p>
+   *  Language code ({@link TranslationFile#UNKNOWN_PART TranslationFile#UNKNOWN_PART}) will be used if unknown
+   * </p>
+   */
+  @Nullable
+  protected Map<String, Map<String, TranslationFile>> translationFileFromLanguageCountryMap =
+    null;
+
+  /**
+   * A map from locales to translations, filled in on-demand as locales are presented.
+   */
+  protected Map<Locale, TranslationMap> localeTranslations = new HashMap<>();
+
+  private final ResourcePatternResolver resourceResolver;
+  private final TranslationConfiguration configuration;
+
+  /**
+   * Get a map from language -&gt; country -&gt; {@link TranslationFile TranslationFile}.
+   * The files are pulled from the filesystem based on the provided
+   * {@link TranslationConfiguration TranslationConfiguration}.
+   *
+   * @return the map of languages to countries to {@link TranslationFile TranslationFile}s
+   */
+  @Nonnull
+  public Map<String, Map<String, TranslationFile>> getFileMap() {
+    if (this.translationFileFromLanguageCountryMap == null) {
+      this.translationFileFromLanguageCountryMap =
+        buildLanguageCountryPatternMap();
+    }
+    return this.translationFileFromLanguageCountryMap;
+  }
+
+  /**
+   * Get the best {@link TranslationMap TranslationMap} associated with the current locale.
+   *
+   * @param locale the locale to find a TranslationMap for
+   * @param fallback the default translation that should be used as a fallback; can be null when
+   *   creating the primary fallback translation
+   * @return the best translation available for the current locale, potentially null if the
+   *   default is null.  It checks for match of language and country, then language, then returns
+   *   default if neither are available
+   */
+  @Nullable
+  public TranslationMap getTranslation(
+    Locale locale,
+    @Nullable TranslationMap fallback
+  ) {
+    return localeTranslations.computeIfAbsent(
+      locale,
+      (Locale missingLocale) -> {
+        log.info("Cache miss on {}; loading map", missingLocale);
+
+        if (
+          this.getFileMap()
+            .containsKey(missingLocale.getLanguage().toLowerCase())
+        ) {
+          Map<String, TranslationFile> languageMap =
+            this.getFileMap().get(missingLocale.getLanguage().toLowerCase());
+
+          TranslationFile baseFile = languageMap.get(
+            TranslationFile.UNKNOWN_PART
+          );
+          TranslationMap baseMap = new TranslationMap(
+            missingLocale,
+            baseFile,
+            fallback
+          );
+
+          if (
+            languageMap.containsKey(missingLocale.getCountry().toLowerCase())
+          ) {
+            return new TranslationMap(
+              missingLocale,
+              languageMap.get(missingLocale.getCountry().toLowerCase()),
+              baseMap
+            );
+          }
+
+          return baseMap;
+        }
+
+        return fallback;
+      }
+    );
+  }
```

Comment:
This method could be 2 times shorter if omit unneeded linebreaks.

Reply:
If you think that it make it less readable, then it could be refactored by extracting methods.

Reply:
I've reformated the class.